DreamJoy_Calc/Form2.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Windows.Forms;
+
+namespace DreamJoy_Calc
+{
+    public partial class Form2 : Form
+    {
+        static private string naam;
+
+        private Mens mens = new Mens(naam);
+        private Papegaai papegaai = new Papegaai(naam);
+        private Kat kat = new Kat(naam);
+
+        public Form2()
+        {
+            InitializeComponent();
+        }
+
+        private void radioButtonKat_CheckedChanged(object sender, EventArgs e)
+        {
+            kat.Naam = naam;
+            if (radioButtonKat.Checked)
+            {
+                listBox1.Enabled = true;
+                buttonEten.Enabled = true;
+                buttonPraten.Enabled = true;
+                buttonStrel.Enabled = true;
+            }
+        }
+
+        private void buttonAanmaken_Click(object sender, EventArgs e)
+        {
+            if (textBoxDier.Text != string.Empty)
+            {
+                naam = textBoxDier.Text;
+                radioButtonKat.Enabled = true;
+                radioButtonMen.Enabled = true;
+                radioButtonPape.Enabled = true;
+            }
+            else
+            {
+                MessageBox.Show("Text box cannot be left blank", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+        }
+
+        private void textBoxDier_TextChanged(object sender, EventArgs e)
+        {
+        }
+
+        private void radioButtonPape_CheckedChanged(object sender, EventArgs e)
+        {
+            papegaai.Naam = naam;
+            if (radioButtonPape.Checked)
+            {
+                listBox1.Enabled = true;
+                buttonEten.Enabled = true;
+                buttonPraten.Enabled = true;
+                buttonStrel.Enabled = true;
+            }
+        }
+
+        private void radioButtonMen_CheckedChanged(object sender, EventArgs e)
+        {
+            mens.Naam = naam;
+            if (radioButtonMen.Checked)
+            {
+                listBox1.Enabled = true;
+                buttonEten.Enabled = true;
+                buttonPraten.Enabled = true;
+                buttonStrel.Enabled = true;
+            }
+        }
+
+        private void listBox1_SelectedIndexChanged(object sender, EventArgs e)
+        {
+        }
+
+        private void buttonPraten_Click(object sender, EventArgs e)
+        {
```

Comment:
Hier is veel duplicate code. Gebruik hiervoor een methode die als input een Dier object gebruikt.

if(radioButtonMen.Checked)
{
WritePraten(man)
}

public void WritePraten(Animal animal)
{
   MessageBox.Show(animal.Praten(listBox1.SelectedItem.ToString()))
}